generic/controllers/generic_controller.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"fmt"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/loaders"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/status"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	//api "sigs.k8s.io/cluster-addons/generic/api/v1alpha1"
+)
+
+var _ reconcile.Reconciler = &GenericReconciler{}
+
+// GenericReconciler reconciles a Generic object
+type GenericReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	declarative.Reconciler
+	GVK     schema.GroupVersionKind
+	Channel string
+}
+
+// +kubebuilder:rbac:groups=addons.x-k8s.io,resources=generics,verbs=get;list;watch;create;update;patch;delete
```

Comment:
I actually don't think we need these rbac permissions.  I guess we want get/list/watch on generics, but you could also move than to main.go.

Reply:
Is there any difference, moving it to `main.go`